File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/misc/TypefaceUtils.java
package com.inspiredandroid.linuxcommandbibliotheca.misc;

import android.content.Context;
import android.graphics.Typeface;

public class TypefaceUtils {

    private static Typeface typeface;

    /**
     * Get typeface by filename from assets main directory
     *
     * @param context
     * @return
     */
    public static Typeface getTypeFace(final Context context) {
        if (typeface == null) {
            typeface = Typeface.createFromAsset(context.getAssets(), "ShareTechMono-Regular.ttf");
        }
        return typeface;
    }
}

File: app/src/main/java/com/inspiredandroid/linuxcommandbibliotheca/misc/TypefaceUtils.kt
package com.inspiredandroid.linuxcommandbibliotheca.misc

import android.content.Context
import android.graphics.Typeface

object TypefaceUtils {

    private var typeface: Typeface? = null

    /**
     * Get typeface by filename from assets main directory
     *
     * @param context
     * @return
     */
    fun getTypeFace(context: Context): Typeface {
        if (typeface == null) {
            typeface = Typeface.createFromAsset(context.assets, "ShareTechMono-Regular.ttf")
        }
        return typeface ?: Typeface.createFromAsset(context.assets, "ShareTechMono-Regular.ttf")
    }
}